### PR TITLE
refactor: feed to only return visible posts

### PR DESCRIPTION
**Pull request by sshanzel**
With the introduction of visible property in `Post` entity, we need to ensure our feeds would only return visible posts.

Updated the fixture that our feeds tests are using and no `visible = false` in the snapshot was shown.

### CODE REVIEW

File: __tests__/cron/__snapshots__/updateTagsStr.ts.snap
Comment (sshanzel) [inline __tests__/cron/__snapshots__/updateTagsStr.ts.snap:32]: This is a direct call from DB in the test not related to any feed.

File: src/schema/posts.ts
Comment (idoshamun) [inline src/schema/posts.ts:222]: I'm not sure we should propagate this field down to the client. Thoughts? cc @rebelchris

**Comment by idoshamun** [comment]
Another idea of implementation, instead of adding a where clause in many places should we consider a new view of VisiblePost that already queries `deleted = false and visible = true` and we will use it everywhere?

**Review by rebelchris — CHANGES_REQUESTED** [review]
Yeah think a new view entity would work really well here.
As long as we also propagate it to Tinybird.

Let me know if you want a quick catch-up on this @sshanzel

File: src/graphorm/graphorm.ts
Comment (sshanzel) [inline src/graphorm/graphorm.ts:310]: @idoshamun I encountered a bug where the expected entity was not used from the query (`from` property) since the table name used for the alias was from the `fromMetadata` hence, pulling the data from the reference entity instead of the view. Let me know if you would want this implemented differently.

Comment (idoshamun) [reply src/graphorm/graphorm.ts:310]: I probably missed this one. You can change it to `this.getMetadata(ctx, type)`

Comment (sshanzel) [reply src/graphorm/graphorm.ts:310]: Cool. Thank you for the confirmation.

File: src/schema/feeds.ts
Comment (sshanzel) [inline src/schema/feeds.ts:916]: Had to convert it to something simpler, the View seems to include a lot of its properties without telling, so we will have to include those columns in the `group by` but there were too many of them so I have tried moving away from using group by.

File: src/entity/posts/ActivePost.ts
Comment (idoshamun) [inline src/entity/posts/ActivePost.ts:16]: Do we really need to define all these fields? I think we can get away without it

File: src/graphorm/graphorm.ts
Comment (idoshamun) [reply src/graphorm/graphorm.ts:310]: I requested to reverted this part to whatever there's in main now. Sorry I wasn't clear before 😅

File: src/schema/feeds.ts
Comment (idoshamun) [reply src/schema/feeds.ts:916]: This is not good. The group by is here to make sure we don't get duplicates. Let's keep the query as it is and try to fix it in another way

File: src/schema/submissions.ts
Comment (idoshamun) [inline src/schema/submissions.ts:164]: I think we should keep querying Post here

File: src/common/feedGenerator.ts
Comment (idoshamun) [inline src/common/feedGenerator.ts:221]: This is not needed any more as well right?

File: src/schema/feeds.ts
Comment (sshanzel) [reply src/schema/feeds.ts:916]: Oh, but there is no duplicate post in the post table, right? This is not an aggregated query (only became one due to the count + filter) so there's no need for us to group it. These selects were only to identify whether the user is the author and/or the scout.

File: src/entity/posts/ActivePost.ts
Comment (sshanzel) [reply src/entity/posts/ActivePost.ts:16]: I agree, just not sure right now which ones are needed and had some trouble with some missing columns earlier so I stashed them all in.

File: src/schema/submissions.ts
Comment (sshanzel) [reply src/schema/submissions.ts:164]: It will not return the existing deleted post anymore since the Post entity in the graphorm will be going to use the new View entity, so it would think it was never in our DB.

File: src/common/feedGenerator.ts
Comment (sshanzel) [reply src/common/feedGenerator.ts:221]: Oh, right. Will do a whole cleanup for any existing implementations, tomorrow.

File: src/migration/1678365541578-ActivePostView.ts
Comment (sshanzel) [reply src/migration/1678365541578-ActivePostView.ts:8]: I think that's alright, we will use graphorm most likely to serve the client. Internally in the API, we can just run queries directly on the repositories as we already have direct access to all the data. One example is in the `submission.ts` changes.

Comment (idoshamun) [reply src/migration/1678365541578-ActivePostView.ts:8]: But in the case I mention we use graphorm. So the post page uses the `post` query to fetch data of the given post and if it's a share post, it also asks for the referred post. In the case we ignore unknown posts, the result will be empty

File: src/schema/feeds.ts
Comment (idoshamun) [reply src/schema/feeds.ts:916]: Ok you're right!

File: src/schema/submissions.ts
Comment (sshanzel) [reply src/schema/submissions.ts:164]: Right. I think we will have to make changes to the `Post` entity to include the `url` and `canonicalUrl` since it is currently available on `ArticlePost` only. Let me update those quickly.

Comment (sshanzel) [reply src/schema/submissions.ts:164]: I think that is what we have here, we are querying directly to the DB by using the typeorm. That's why we had to introduce the change.

File: src/migration/1678365541578-ActivePostView.ts
Comment (sshanzel) [reply src/migration/1678365541578-ActivePostView.ts:8]: Oh, right. We should then remove the additional checks such as the source and let the `visible` property handle whether the post should be deemed valid to be returned. Wow, nice catch right there.

File: src/entity/posts/ActivePost.ts
Comment (sshanzel) [inline src/entity/posts/ActivePost.ts:13]: Looks like we only need two columns for this class! 🚢

File: src/common/feedGenerator.ts
Comment (sshanzel) [reply src/common/feedGenerator.ts:221]: Did clean-ups now, and just kept the source condition since as discussed, we will remove same condition from the View condition.

**Comment by sshanzel** [comment]
@rebelchris would love to hear what you think! 🚢

File: src/schema/submissions.ts
Comment (rebelchris) [inline src/schema/submissions.ts:176]: This feels a bit duplicate from the one above.
Let's maybe see if it can be uniformed one way?

**Review by rebelchris — CHANGES_REQUESTED** [review]
Tested locally, but for instance source feed builder is not actively using this.
We need to be very careful rolling this out as it has high impact on existing feeds.

See here:
Screenshot 2023-03-10 at 13 52 10

Specific code:
Screenshot 2023-03-10 at 13 51 28

But maybe we should add test-cases with invisible posts and debug backwards?
Just for future-proofing.

File: src/schema/submissions.ts
Comment (sshanzel) [reply src/schema/submissions.ts:176]: This function is using graphorm, the change introduced above was using typeorm.

**Comment by sshanzel** [comment]
> Specific code:

That is using the `feedResolver` which uses the `Post` entity in the graphorm which will use the `ActivePost` as defined. Let me check further on how this happens.

**Review by rebelchris — APPROVED** [review]
Nice sorted locally now.
Just want to get Ido's approval on his comment before we merge.

File: src/cron/exportToTinybird.ts
Comment (idoshamun) [inline src/cron/exportToTinybird.ts:35]: This needs to use the real table

```suggestion
       FROM "post"
```

File: src/cron/exportToTinybird.ts
Comment (idoshamun) [inline src/cron/exportToTinybird.ts:36]: Let's add here a condition that the post is visible

File: src/entity/posts/utils.ts
Comment (idoshamun) [inline src/entity/posts/utils.ts:43]: This ignores the new view. Did you mean `ActivePost`?

**Review by idoshamun — APPROVED** [review]

**Pull request merged**
